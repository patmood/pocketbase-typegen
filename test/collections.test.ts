import {
  createCollectionEnum,
  createCollectionRecords,
  createCollectionResponses,
} from "../src/collections"

describe("createCollectionEnum", () => {
  it("creates enum of collection names", () => {
    const names = ["book", "magazine"]
    expect(createCollectionEnum(names)).toMatchSnapshot()
  })
})

describe("createCollectionRecords", () => {
  it("creates mapping of collection name to record type", () => {
    const names = ["book", "magazine"]
    expect(createCollectionRecords(names)).toMatchSnapshot()
  })
})

describe("createCollectionResponses", () => {
  it("creates mapping of collection name to response type", () => {
    const names = ["book", "magazine"]
    expect(createCollectionResponses(names)).toMatchSnapshot()
  })
})
