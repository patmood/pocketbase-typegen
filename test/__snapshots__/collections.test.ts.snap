// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCollectionEnum creates enum of collection names 1`] = `
"export enum Collections {
	Book = "book",
	Magazine = "magazine",
}"
`;

exports[`createCollectionRecords creates mapping of collection name to record type 1`] = `
"export type CollectionRecords = {
	book: BookRecord
	magazine: MagazineRecord
}"
`;

exports[`createCollectionResponses creates mapping of collection name to response type 1`] = `
"export type CollectionResponses = {
	book: BookResponse
	magazine: MagazineResponse
}"
`;

exports[`createTypedPocketBase creates typed variant of PocketBase client 1`] = `
"export type TypedPocketBase = PocketBase & {
	collection(idOrName: 'book'): RecordService<BookResponse>
	collection(idOrName: 'magazine'): RecordService<MagazineResponse>
}"
`;
